include/deal.II/lac/block_vector_base.h

```diff
@@ -2264,6 +2262,17 @@ void BlockVectorBase<VectorType>::extract_subvector_to (ForwardIterator
     }
 }
 
+
+template <typename VectorType>
+inline
+bool BlockVectorBase<VectorType>::supports_distributed_data() const
+{
+  // Create a dummy vector of type VectorType to check
+  // supports_distributed_data even if BlockVectorBase contains no block
+  VectorType dummy;
+  return dummy.supports_distributed_data();
```

Comment:
@tamiko -- `BlockVector` actually knows the exact type of the underlying vector types for the individual blocks, because it allocates them. They're equal to the template type. In other words, we know for sure that when @Rombur uses
```
  VectorType dummy;
  return dummy.supports_distributed_data();
```
that dummy has the right type.

But there is still the original point I don't understand and that I haven't understood from @Rombur 's comments. If you argue that this now needs to be a function rather than a static variable, then this only makes sense if one class may either return `true` or `false`, depending on what its current internal state is. Otherwise, if it does not depend on the internal state (i.e., the values of member variables), then it could as well be a static const variable. But if, indeed, the returned value may depend on the state of the variables, then necessarily the `BlockVector` class cannot just create an empty object, but actually needs to query the individual blocks. Isn't this right?

(The other interpretation is that you want to access the variable through the *base class* but you don't know what derived class you have. In that case, the function will be abstract virtual. I may be missing this, but I don't actually see a place where you use this right now. Do you have a use case in mind where you only get a pointer-to-base, but still need to know whether the vector supports distributed data?)